internal/gouging/gouging.go

```diff
@@ -113,8 +106,59 @@ func (gc checker) Check(hs *rhpv2.HostSettings, pt *rhpv3.HostPriceTable) api.Ho
 	}
 }
 
+// TODO: write tests
+func (gc checker) CheckV2(hs rhp.HostSettings) (gb api.HostGougingBreakdown) {
+	prices := hs.Prices
+	gs := gc.settings
+
+	// upload gouging
+	var uploadErrs []error
+	if prices.StoragePrice.Cmp(gs.MaxStoragePrice) > 0 {
+		uploadErrs = append(uploadErrs, fmt.Errorf("%v: storage price exceeds max storage price: %v > %v", ErrPriceTableGouging, prices.StoragePrice, gs.MaxStoragePrice))
+	}
+	if prices.IngressPrice.Cmp(gs.MaxUploadPrice) > 0 {
+		uploadErrs = append(uploadErrs, fmt.Errorf("%v: ingress price exceeds max upload price: %v > %v", ErrPriceTableGouging, prices.IngressPrice, gs.MaxUploadPrice))
+	}
+	gb.UploadErr = errsToStr(uploadErrs...)
+	if gougingErr := errsToStr(uploadErrs...); gougingErr != "" {
+		gb.UploadErr = fmt.Sprintf("%v: %s", ErrPriceTableGouging, gougingErr)
+	}
+
+	// download gouging
+	if prices.EgressPrice.Cmp(gs.MaxDownloadPrice) > 0 {
+		gb.DownloadErr = fmt.Sprintf("%v: egress price exceeds max download price: %v > %v", ErrPriceTableGouging, prices.EgressPrice, gs.MaxDownloadPrice)
```

Comment:
should `renterd` adopt egress/ingress in v2?

Reply:
Maybe for v3.0. Theoretically things like upload gouging consist of more than just the egress/ingress prices which is a bit awkward.